src/search/pdbs/rcg.cc

```diff
@@ -0,0 +1,87 @@
+#include "rcg.h"
+
+#include "../option_parser.h"
+
+#include "../task_proxy.h"
+
+#include "../utils/countdown_timer.h"
+#include "../utils/logging.h"
+#include "../utils/rng.h"
+
+#include <algorithm>
+#include <unordered_set>
+
+using namespace std;
+
+namespace pdbs{
+static bool time_limit_reached(
```

Comment:
This looks like it belongs to `utils::CountdownTimer`. Is this the right place to implement this?

Reply:
I don't think it should be a method of `utils::CountdownTimer`: there is already `is_expired` for this and I just want to have the added output. It could be a function in the `utils::countdown_timer`, if you prefer.

Reply:
Yes, I think it makes more sense there. You could also extend `is_expired` by adding an optional `logger` argument.

Reply:
I looked into the usages of `utils::CountdownTimer::is_expired()` and would argue to not add anything to `utils::coutdown_timer.h` that has to do with output. All users of `utils::CountdownTimer::is_expired()` print their own message and possibly do additional things in the if-clause. We could pass both `Log logger` and `string message` to the timer class(es), but I don't really think it belongs there.

Reply:
OK, fine with me. Maybe you could go in the other direction and make the log message here more specific as well. What kind of time limit was reached, what does this mean for the computation? That sort of thing.